### PR TITLE
Fix allocation bug instance connection error

### PR DESCRIPTION
**Pull request by c-bordon**
close: https://github.com/wazuh/wazuh-qa/issues/5198

A connectivity validation method is created after provisioning, this validation is carried out through `SSH` or `WINRM` depending on the system. When the module finishes executing it is because the machine is provisioned and has connectivity, therefore, it is ready to use.

Several tests were carried out and they were used to correct some errors in the Windows userData and the general Windows configuration. As well as in the process of eliminating Windows instances.

The validation process has an automatic rollback, that is, if after 15 retries with a time between attempts of 60 seconds the instance does not connect, a rollback is automatically performed and the instance is deleted. This rollback can be disabled with the `--rollback` argument set to `False`

### Test Windows desktop 11: https://github.com/wazuh/wazuh-qa/issues/5198#issuecomment-2057674244

### CODE REVIEW

File: deployability/modules/allocation/static/specs/os.yml [hunk 1]
```diff
@@ -367,24 +367,24 @@ aws:
...
-    user: Administrator
-  windows-desktop-11-amd64:
-    ami: ami-09d8cef159442d5b0
-    zone: us-east-1
-    user: Jenkins
+    user: wazuh-user
+  #windows-desktop-11-amd64: Issue related https://github.com/wazuh/wazuh-automation/issues/1582
+  #  ami: ami-0115c7e59b426bffb
+  #  zone: us-east-1
+  #  user: Jenkins
```
Comment (rauldpm) [inline deployability/modules/allocation/static/specs/os.yml:374]: Remove it and create an issue with DTT2 objective please

**Review by rauldpm — CHANGES_REQUESTED** [review]

File: deployability/modules/allocation/allocation.py [hunk 2]
```diff
@@ -3,6 +3,8 @@
 # This program is a free software; you can redistribute it and/or modify it under the terms of GPLv2
 
 import yaml
+import paramiko, logging, time
+import winrm
```
Comment (rauldpm) [inline deployability/modules/allocation/allocation.py:8]: Add `pywinrm` to `requirements.txt` file with a version

```
[2024-04-16 20:06:03] [DEBUG] [57246] [ThreadPoolExecutor-1_0] [workflow_engine]: Running task "allocate-manager-linux-ubuntu-22.04-amd64" with arguments: ['modules/allocation/main.py', '--action=delete', '--track-output=/tmp/dtt1-poc/manager-linux-ubuntu-22.04-amd64/track.yaml']
[2024-04-16 20:06:03] [ERROR] [57246] [ThreadPoolExecutor-1_0] [workflow_engine]: [allocate-manager-linux-ubuntu-22.04-amd64] Task failed with error: Error executing process task Traceback (most recent call last):
  File "/home/xxx/Documents/Wazuh/Repositories/wazuh-qa/wazuh-qa/deployability/modules/allocation/main.py", line 12, in <module>
    from modules.allocation import Allocator
  File "/home/xxx/Documents/Wazuh/Repositories/wazuh-qa/wazuh-qa/deployability/modules/__init__.py", line 6, in <module>
    from .allocation import Allocator
  File "/home/xxx/Documents/Wazuh/Repositories/wazuh-qa/wazuh-qa/deployability/modules/allocation/__init__.py", line 5, in <module>
    from .allocation import Allocator
  File "/home/xxx/Documents/Wazuh/Repositories/wazuh-qa/wazuh-qa/deployability/modules/allocation/allocation.py", line 7, in <module>
    import winrm
ModuleNotFoundError: No module named 'winrm'
```

When I tried to add it I got this error

```
ERROR: Could not find a version that satisfies the requirement winrm (from versions: none)
ERROR: No matching distribution found for winrm
```

**Review by rauldpm — CHANGES_REQUESTED** [review]

File: deployability/modules/allocation/static/specs/os.yml (hunk 1 above)
Comment (teddytpc1) [reply deployability/modules/allocation/static/specs/os.yml:374]: Issue created: https://github.com/wazuh/wazuh-automation/issues/1582

Comment (c-bordon) [reply deployability/modules/allocation/static/specs/os.yml]: Done: 1ffd41c6b

File: deployability/modules/allocation/allocation.py (hunk 2 above)
Comment (c-bordon) [reply deployability/modules/allocation/allocation.py:8]: Done: 1ffd41c6b

**Review by rauldpm — APPROVED** [review]

**Review by fcaffieri — APPROVED** [review]

**Pull request merged**